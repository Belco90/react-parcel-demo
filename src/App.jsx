import React, { useState, useEffect } from 'react'
import logo from './logo.svg'
import './App.css'

export const App = () => {
  const [secondsOpen, setSecondsOpen] = useState(0)
  const [clicksCount, setClicksCount] = useState(0)

  useEffect(() => {
    const timer = setInterval(
      () => setSecondsOpen((prevSecondsOpen) => prevSecondsOpen + 1),
      1000
    )
    return () => clearTimeout(timer)
  }, [])

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="React logo" />
        <h1>Hello Parcel + React!</h1>
        <p>
          Edit <code>App.jsx</code> and save to test HMR updates.
        </p>
        <p>
          Page has been open for <code>{secondsOpen}</code> seconds.
        </p>
        <p>
          <button
            type="button"
            onClick={() => setClicksCount((count) => count + 1)}
          >
            Count is: {clicksCount}
          </button>
        </p>

        <p>
          <a
            className="App-link"
            href="https://beta.reactjs.org/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
          {' | '}
          <a
            className="App-link"
            href="https://parceljs.org/docs/"
            target="_blank"
            rel="noopener noreferrer"
          >
            Parcel Docs
          </a>
        </p>
      </header>
    </div>
  )
}
