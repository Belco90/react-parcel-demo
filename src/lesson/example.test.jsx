import React from 'react'
import { render, screen } from '@testing-library/react'
import { App } from '../App'

it('should render a basic demo', () => {
  render(<App />)
  const banner = screen.getByRole('banner')
  const heading = screen.getByRole('heading', { name: 'Hello Parcel + React!' })
  expect(banner.firstChild).toHaveAttribute('alt', 'React logo')
  expect(heading.closest('header')).toHaveClass('App-header')
})
