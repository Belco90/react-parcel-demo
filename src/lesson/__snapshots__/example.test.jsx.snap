// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the basic Pokémon info 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
  </header>
</div>
`;
